src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-features-count/ProteinFeaturesCount.tsx

```diff
@@ -14,79 +14,59 @@
 import annotationsIcon from 'static/img/entity-viewer/icon_protein_annotations.svg';
 
-import transcriptListStyles from 'src/content/app/entity-viewer/gene-view/components/default-transcripts-list/DefaultTranscriptsList.scss';
 import styles from './ProteinFeaturesCount.scss';
 
 type ProteinFeaturesCountProps = {
-  transcriptId: string;
+  proteinStats: ProteinStats;
 };
 
 const ProteinFeaturesCount = (props: ProteinFeaturesCountProps) => {
-  const [proteinStats, setProteinStats] = useState<ProteinStats | null>(null);
-
-  useEffect(() => {
-    fetchProteinSummaryStats(props.transcriptId).then((response) => {
-      setProteinStats(response);
-    });
-  }, [props.transcriptId]);
-
-  if (!proteinStats) {
-    return null;
-  }
-
-  const midStyles = classNames(transcriptListStyles.middle, styles.middle);
+  const { proteinStats } = props;
 
   return (
-    <div className={transcriptListStyles.row}>
-      <div className={transcriptListStyles.left}>PDBe-KB P51587</div>
-      <div className={midStyles}>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={structuresIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>
-            {proteinStats.structuresCount}
-          </div>
-          <div className={styles.featureText}>Structures</div>
+    <div>
+      <div className={styles.feature}>
+        <div className={styles.featureImg}>
+          <img src={structuresIcon} alt="" />
         </div>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={ligandsIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>{proteinStats.ligandsCount}</div>
-          <div className={styles.featureText}>Ligands</div>
+        <div className={styles.featureCount}>
+          {proteinStats.structuresCount}
+        </div>
+        <div className={styles.featureText}>Structures</div>
+      </div>
+      <div className={styles.feature}>
+        <div className={styles.featureImg}>
+          <img src={ligandsIcon} alt="" />
         </div>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={interactionsIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>
-            {proteinStats.interactionsCount}
-          </div>
-          <div className={styles.featureText}>Interactions</div>
+        <div className={styles.featureCount}>{proteinStats.ligandsCount}</div>
+        <div className={styles.featureText}>Ligands</div>
+      </div>
+      <div className={styles.feature}>
+        <div className={styles.featureImg}>
+          <img src={interactionsIcon} alt="" />
+        </div>
+        <div className={styles.featureCount}>
+          {proteinStats.interactionsCount}
+        </div>
+        <div className={styles.featureText}>Interactions</div>
+      </div>
```

Comment:
Wouldn't it be better to create a function for this:
```javascript
const renderFeature = (props) => {
    <div className={styles.feature}>
        <div className={styles.featureCount}>
            <div className={styles.featureImg}>
                <img src={props.icon} alt="" />
            </div>
        </div>
        <div className={styles.featureCount}>
            {props.featureCount}
        </div>
        <div className={styles.featureText}>{props.featureText}</div>
    </div>

}


```

Reply:
This is done now. I created a separate component inside the same file called `FeatureCount` to render this.